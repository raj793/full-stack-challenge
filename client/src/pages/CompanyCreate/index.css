.company-create-container {
  display: flex;
  flex-direction: column;
}

.company-create-input {
  background-color: #FFFFFF;
}

.form-container {
  display: flex;
  flex-direction: column;
  height: 100%; 
  width: 100%; 
  justify-content: space-evenly;
  align-items: center; 
  background-color: #FFFFFF;
}

.heading {
  font-family: monospace;
  background-color: #FFFFFF;
  font-size: 20px;
  margin: 10px;
}

.input-name {
  width: 100%;
  background-color: #FFFFFF;
  border: 1px #000000 solid;
  height: 30px;
}

.input-group-container {
  width: 100%;
  justify-content: space-between;
  display: flex;
  flex-direction: row;
  margin: 10px;
  background-color: #FFFFFF;
}

.input-group-item {
  width: 32%;
  background-color: #FFFFFF;
  border: 1px #000000 solid;
  height: 30px;
}

.input-description {
  resize: none;
  width: 100%;
  background-color: #FFFFFF;
  border: 1px #000000 solid;
  height: 100px;
}

.input-error {
  border: 3px red solid;
}