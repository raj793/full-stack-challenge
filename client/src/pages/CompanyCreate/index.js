import React, { useState } from 'react';
import clsx from 'clsx';
import Api from '../../apis';
import './index.css';

const CompanyCreate = ({onClose, formModelData, update = false}) => {
  const apis = new Api();
  console.log(formModelData, update);
  const formModel = update ? {
    name: formModelData.name,
    city: formModelData.city,
    state: formModelData.state,
    description: formModelData.description,
    founded: formModelData.founded.split('T')[0]
  }:{
    name: '',
    city: '',
    state: '',
    founded: '',
    description: ''
  };

  const [formData, setFormData] = useState(formModel);
  const [errors, setErrors] = useState(formModel);
  const [isSaving, setSaving] = useState(false);

  const setFormValue = (e) => {
    const fd = {...formData};
    fd[e.target.name] = e.target.value;
    setFormData(fd);
  }

  const createNewCompany = async () => {
    if(formData.name && formData.city && formData.state && formData.description) {
      setSaving(true);
      update ? 
      await apis.updateCompanyById(formModelData.id, {
        name: formData.name,
        city: formData.city,
        state: formData.state,
        description: formData.description,
        founded: formData.founded
      })
      : await apis.createCompany({
        name: formData.name,
        city: formData.city,
        state: formData.state,
        description: formData.description,
        founded: formData.founded ? formData.founded: null
      });
      setSaving(false);
      onClose && onClose();
    }
    else {
      setErrors({
        name: !formData.name,
        city: !formData.city,
        state: !formData.state,
        description: !formData.description
      })
    }
  }

  return (
    <div className="form-container">
      <span className="heading">Create a new company</span>
      <input 
        className={clsx({["input-name"]: true, ["input-error"]: errors.name && !update})} 
        name="name" 
        onChange={setFormValue} 
        value={formData.name} 
        placeholder=" Company Name *Required"
      />
      <div className="input-group-container">
        <input 
          className={clsx({["input-group-item"]: true, ["input-error"]: errors.city && !update})} 
          name="city" 
          onChange={setFormValue} 
          value={formData.city} 
          placeholder=" City *Required"
        />
        <input 
          className={clsx({["input-group-item"]: true, ["input-error"]: errors.state && !update})}
          name="state"
          onChange={setFormValue}
          value={formData.state}
          placeholder=" State *Required"
        />
        <input
          className="input-group-item"
          name="founded"
          onChange={setFormValue}
          value={formData.founded}
          onFocus={e => e.currentTarget.type = 'Date'}
          placeholder=" Founded Date"
        />
      </div>
      <textarea 
        className={clsx({["input-description"]: true, ["input-error"]: errors.description && !update})} 
        name="description" 
        onChange={setFormValue} 
        value={formData.description} 
        placeholder=" Description *Required"
      />
      <div className='save-button' onClick={createNewCompany}>{isSaving? "SAVING": "SAVE"}</div>
    </div>
  );
}

export default CompanyCreate;