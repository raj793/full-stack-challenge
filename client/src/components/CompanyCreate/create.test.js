import React from 'react';
import renderer from 'react-test-renderer';
import CompanyCreate from './index';

test('Company Create Snapshot Test', () => {
  const formData = {
    name: "Name",
    city: "City",
    state: "State",
    description: "Description"
  }
  const component = renderer.create(
    <CompanyCreate formModelData={formData} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});