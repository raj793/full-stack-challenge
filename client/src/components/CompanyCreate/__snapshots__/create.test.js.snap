// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company Create Snapshot Test 1`] = `
<div
  className="form-container"
>
  <span
    className="heading"
  >
    Create
     a company
  </span>
  <input
    className="input-name"
    name="name"
    onChange={[Function]}
    placeholder=" Company Name *Required"
    value=""
  />
  <div
    className="input-group-container"
  >
    <input
      className="input-group-item"
      name="city"
      onChange={[Function]}
      placeholder=" City *Required"
      value=""
    />
    <input
      className="input-group-item"
      name="state"
      onChange={[Function]}
      placeholder=" State *Required"
      value=""
    />
    <input
      className="input-group-item"
      name="founded"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder=" Founded Date"
      value=""
    />
  </div>
  <textarea
    className="input-description"
    name="description"
    onChange={[Function]}
    placeholder=" Description *Required"
    value=""
  />
  <div
    className="save-button"
    onClick={[Function]}
  >
    SAVE
  </div>
</div>
`;
